File: cpp/524_通过删除字母匹配到字典里最长单词_medium.cpp
class Solution {
public:
    string findLongestWord(string s, vector<string>& dictionary) {
        string ans = "";
        for (auto &t : dictionary) {
            int i = 0, j = 0;
            while (i < t.size() && j < s.size()) {
                if (t[i] == s[j]) ++i;
                ++j;
            }
            if (i == t.size()) {
                if (ans.size() == t.size())
                    ans = t > ans ? ans : t;
                else
                    ans = t.size() > ans.size() ? t : ans;
            }
        }
        return ans;
    }
};

File: python/524_通过删除字母匹配到字典里最长单词_medium.py
class Solution:
    def findLongestWord(self, s: str, dictionary: List[str]) -> str:
        ans = ""
        for t in dictionary:
            i, j = 0, 0
            while i < len(t) and j < len(s):
                if t[i] == s[j]:
                    i += 1
                j += 1
            if i == len(t):
                if len(t) == len(ans):
                    ans = t if t < ans else ans
                else:
                    ans = t if len(t) > len(ans) else ans
        return ans